src/bp/core/services/bot-service.ts

```diff
@@ -451,4 +465,75 @@ export class BotService {
     BotService._mountedBots.forEach((isMounted, bot) => isMounted && bots.push(bot))
     return bots
   }
+
+  public async listRevisions(botId: string): Promise<string[]> {
+    const globalGhost = this.ghostService.global()
+
+    let stageID = ''
+    if (await this.workspaceService.hasPipeline()) {
+      const botConfig = await this.configProvider.getBotConfig(botId)
+      stageID = botConfig.pipeline_status.current_stage.id
+    }
+
+    const revisions = await globalGhost.directoryListing(REVISIONS_DIR)
+
+    return revisions.filter(rev => rev.includes(botId) && rev.includes(stageID)).sort((revA, revB) => {
+      const dateA = revA.split('::')[1].replace('.tgz', '')
```

Comment:
On windows, a filename can't have :: in its name, I suggest using another. Not allowed: \/:*?"<>|.  Maybe it won't be an issue on postgres, but saving revisions on the filesystem won't work. How about using + instead? We just need to make sure it can be used in a bot id or stage name